src/Domain/Tags.hpp

```diff
@@ -252,15 +252,21 @@ struct unmap_interface_args;
 
 template <>
 struct unmap_interface_args<true> {
+  template <typename T>
+  using f = T;
+
   template <size_t VolumeDim, typename T>
-  static constexpr decltype(auto) apply(
-      const ::Direction<VolumeDim>& /*direction*/, const T& arg) noexcept {
+  static constexpr const T& apply(const ::Direction<VolumeDim>& /*direction*/,
+                                  const T& arg) noexcept {
```

Comment:
What's this change about?  Didn't the return type work out to `const T&` already?

Reply:
I find it easier if in these simple cases the type is more explicit and was useful in debugging the template issues to convince myself the code was behaving as expected :)